contracts/tokenomics/generator/src/contract.rs

```diff
@@ -1197,6 +1201,83 @@ fn send_orphan_proxy_rewards(
         .add_attribute("amount", amount))
 }
 
+/// ## Description
+/// Sets the reward proxy contract for the pool. Returns an [`ContractError`] on failure, otherwise
+/// returns the [`Response`] with the specified attributes if the operation was successful.
+fn update_pool_config(
+    mut deps: DepsMut,
+    env: Env,
+    lp_token: String,
+    proxy: String,
+) -> Result<Response, ContractError> {
+    let lp_addr = addr_validate_to_lower(deps.api, &lp_token)?;
+    let proxy_addr = addr_validate_to_lower(deps.api, &proxy)?;
+
+    let cfg = CONFIG.load(deps.storage)?;
+
+    if !cfg.allowed_reward_proxies.contains(&proxy_addr) {
+        return Err(ContractError::RewardProxyNotAllowed {});
+    }
+
+    let mut pool_info = POOL_INFO.load(deps.storage, &lp_addr)?;
+    pool_info.reward_proxy = Some(proxy_addr);
+
+    accumulate_rewards_per_share(deps.branch(), &env, &lp_addr, &mut pool_info, &cfg, None)?;
+
+    POOL_INFO.save(deps.storage, &lp_addr, &pool_info)?;
+
+    Ok(Response::new().add_attributes(vec![
+        attr("action", "update_pool_config"),
+        attr("new_proxy", proxy),
+    ]))
+}
+
+/// ## Description
+/// Adds or removes the proxy contracts to the list of allowed. Returns an [`ContractError`] on failure
+fn update_proxies(
+    deps: DepsMut,
+    add: Option<Vec<String>>,
+    remove: Option<Vec<String>>,
+) -> Result<Response, ContractError> {
+    if add.is_none() && remove.is_none() {
+        return Err(ContractError::Std(StdError::generic_err(
+            "Need to provide add or remove parameters",
+        )));
+    }
+
+    let mut cfg = CONFIG.load(deps.storage)?;
+
+    // remove old proxy
+    if let Some(remove_proxies) = remove {
+        for remove_proxy in remove_proxies {
+            let remove_proxy_addr = addr_validate_to_lower(deps.api, &remove_proxy)?;
+            let index = cfg
+                .allowed_reward_proxies
+                .iter()
+                .position(|x| *x == remove_proxy_addr)
```

Comment:
We can just use x.as_str() == remove_proxy_addr.as_str() in the position closure without validating